java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/specification/active/questionnaire/QuestionnaireDataTopic.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2017 King's College London and The Hyve
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.radarcns.schema.specification.active.questionnaire;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.radarcns.schema.specification.DataTopic;
+
+import javax.validation.constraints.NotEmpty;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * TODO.
+ */
+public class QuestionnaireDataTopic extends DataTopic {
+    public enum RadarSourceTypes {
+        PHQ8
```

Comment:
PHQ8 enum: should these types be derived from the specification catalogue TYPE (otherwise do we have to define here again)?

Reply:
For extensibility, the specifications use a `string` to define the type of questionnaire or data source. That way, someone can add specifications with types not present in RADAR. For validation, we do check if the given types are valid within RADAR. Concretely:

- Within RADAR-CNS, we presently only support the PHQ8 questionnaire. For any other value, the validator will notify the user that this is not a RADAR questionnaire. If another type is needed, it should be added to this enum.
- Outside RADAR-CNS, for example Thinc-It may want to add a custom `THINCIT-Q1` type. If the specification itself was an enum, this would not parse correctly. Now it will parse, but it will just not be recognised as a RADAR questionnaire.

Reply:
ok clear now.